ui/app/components/app/network.js

```diff
@@ -21,160 +54,112 @@ export default class Network extends Component {
   }
 
   render () {
-    const context = this.context
-    const networkNumber = this.props.network
+    const {
+      disabled,
+      network: networkNumber,
+      onClick,
+    } = this.props
+
+    const {
+      t,
+    } = this.context
+
     let providerName, providerNick, providerUrl
-    try {
+    if (this.props.provider) {
```

Comment:
nit: would it make sense to destructure `provider` from `this.props` like what you've done with `disabled`?

Reply:
Good point, I'll clean this up